HiP-DataStore/Core/ReadModel/CacheDatabaseManager.cs

```diff
@@ -248,6 +229,90 @@ private void ApplyEvent(IEvent ev)
                     _db.GetCollection<ScoreRecord>(ResourceType.ScoreRecord.Name).InsertOne(newScoreRecord);
                     break;
             }
+
+            if (ev is ICreateEvent createEvent)
+                UpdateReferences((createEvent.GetEntityType(), createEvent.Id), createEvent.GetReferences());
```

Comment:
Why Update References? This is new object without references. And you clear it in UpdateReferences. Maybe just use AddReferences()